pytorch_lightning/tuner/auto_gpu_select.py

```diff
@@ -12,9 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import torch
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
 
 
 def pick_multiple_gpus(nb):
+    if nb == 0:
+        raise MisconfigurationException(
+            "you selected `auto_select_gpus=True`. but did not set gpu resource `gpus=0`. Please set a GPU device."
```

Comment:
The wording is a bit odd, maybe use lezwon's suggested message?

`auto_select_gpus=True, gpus=0 is not a valid configuration. Please select a valid number of GPU resources when using auto_select_gpus.`